spring-kafka/src/main/java/org/springframework/kafka/config/AbstractKafkaListenerEndpoint.java

```diff
@@ -326,6 +326,7 @@ public void setReplyTemplate(KafkaTemplate<?, ?> replyTemplate) {
 	@SuppressWarnings("unchecked")
 	public void setRecordFilterStrategy(RecordFilterStrategy<? super K, ? super V> recordFilterStrategy) {
 		this.recordFilterStrategy = (RecordFilterStrategy<K, V>) recordFilterStrategy;
+		setAckDiscarded(true);
```

Comment:
Looking deeper, I don't think this fix is correct.
See more further:
```
	/**
	 * Create an instance with the supplied strategy and delegate listener.
	 * @param delegate the delegate.
	 * @param recordFilterStrategy the filter.
	 * @param ackDiscarded true to ack (commit offset for) discarded messages when the
	 * listener is configured for manual acks.
	 */
	public FilteringMessageListenerAdapter(MessageListener<K, V> delegate,
			RecordFilterStrategy<K, V> recordFilterStrategy, boolean ackDiscarded) {
```
Therefore it is up to end-user to decide if `ackDiscarded` or not.

So, what the problem we are trying to fix here, please?

Reply:
I see 👍 

How about fix the java docs of `setAckDiscarded `'s `Set to true if the {@link #setRecordFilterStrategy(RecordFilterStrategy)} is in use.` ?

For example, `Set to true if the {@link #setRecordFilterStrategy(RecordFilterStrategy)} is in use and ack discarded messages`?

Reply:
How about:
```
Set to true if the {@link #setRecordFilterStrategy(RecordFilterStrategy)} should ack discarded messages.
```
?

Reply:
It's more suitable. (sorry i'm not good at english well🥲 ).
may i fix it?

Reply:
Thanks a lot 🙇‍♂️ 
I updated It!